planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h

```diff
@@ -336,8 +336,19 @@ class PlanningSceneMonitor : private boost::noncopyable
   /** @brief This function is called every time there is a change to the planning scene */
   void triggerSceneUpdateEvent(SceneUpdateType update_type);
 
-  /** \brief Wait until all pending scene updates with timestamps < t are incorporated */
-  void syncSceneUpdates(const ros::Time &t = ros::Time::now());
+  /** \brief Wait for robot state to become more recent than t.
+   *
+   * If there is no state monitor active, there will be no scene updates.
+   * Hence, you can specify a timeout to wait for those updates. Default is 1s.
+   */
+  bool waitForCurrentRobotState(const ros::Time &t, double wait_time = 1.);
+
+  /** \brief Wait current robot state and wait for all pending scene updates to be processed.
+   *
+   * Additionally to waitForCurrentRobotState() this processes also all already received scene updates
+   * as long as wait_time is not exceeded.
+   */
+  bool syncSceneUpdates(const ros::Time &t = ros::Time::now(), double wait_time = 1.);
```

Comment:
On Wed, Aug 17, 2016 at 12:02:41PM -0700, Robert Haschke wrote:
> If scene updates and planning requests are triggered from the very same program,
> the planning request is definitely processed after the update requests, isn't it?.

While this might be true (I'm not sure and it might depend on the order in which the subscriber/publisher are created),
I don't think anyone should rely on this.

> if another source for scene updates is involved. However, in that case the program sending the planning request cannot sync these updates beforehand as well.

Sure, he can hold a PSM depending on `move_group`'s output and wait for expected updates to become visible.
Lousy solution, still, I would always prefer synchronous updates...